spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -66,14 +66,11 @@ else if (tokens.length == 3) {
 							" [<transport>:]<type>:<name>");
 				}
 				Binder<MessageChannel> binder = binderFactory.getBinder(transport);
-				if ("queue".equals(type)) {
+				if ("queue".equals(type)) { // TODO: remove
 					channel = binder.bindDynamicProducer(name, this.producerProperties);
 				}
-				else if ("topic".equals(type)) {
```

Comment:
Ditching the 'topic' prefix makes it a little ambiguous, as '[transport]:[name]' and 'queue:[name]' will take the branch where the `type` becomes `[transport]`. IMO should add another check that the first token is not `queue` before deciding which scheme are we going to use on parsing. Of course, that will cause issues with folks naming a transport `queue`, but that's fine, I would suppose. And `queue` is rather not long lived anyway.